chips/ibex/src/plic.rs

```diff
@@ -0,0 +1,101 @@
+//! Platform Level Interrupt Control peripheral driver.
+
+use kernel::common::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
+use kernel::common::StaticRef;
+
+register_structs! {
+    pub PlicRegisters {
+        /// Interrupt Pending Register
+        (0x000 => pending: [ReadOnly<u32>; 2]),
+        /// Interrupt Source Register
+        (0x008 => source: [ReadWrite<u32>; 2]),
+        /// Interrupt Priority Registers
+        (0x010 => priority: [ReadWrite<u32, priority::Register>; 63]),
+        (0x10C => _reserved0: [ReadWrite<u32>; 61]),
+        /// Interrupt Enable Register
+        (0x200 => enable: [ReadWrite<u32>; 2]),
+        /// Priority Threshold Register
+        (0x208 => threshold: ReadWrite<u32, priority::Register>),
+        /// Claim/Complete Register
+        (0x20C => claim: ReadWrite<u32>),
+        /// MSIP Register
+        (0x210 => msip: ReadWrite<u32>),
+        (0x214 => @END),
+    }
```

Comment:
Can you explain the discrepancy between the offsets here and the offsets in https://docs.opentitan.org/hw/ip/rv_plic/doc? The offsets seem to be 2x what is listed in the documentation.

I imaging is has to do with this bit in the documentation:
> The below register description may not match with top level design. The RV_PLIC in the top level is generated by topgen tool so that the number of interrupt sources may be different.

Reply:
Yeah, the documentation doesn't match what is in the actual hardware. The problem is that they override the register layout in the top level.